src/main/java/com/cloudbees/jenkins/support/impl/JenkinsLogs.java

```diff
@@ -112,9 +109,18 @@ protected void printTo(PrintWriter out) throws IOException {
                                     out.println("N/A");
                                 } else {
                                     try {
-                                        List<LogRecord> records = needHack
-                                                ? SlaveLogFetcher.getLogRecords(computer)
-                                                : computer.getLogRecords();
+                                        List<LogRecord> records = null;
+                                        if (needHack) {
+                                            if (computer.getChannel() != null) {
+                                                Future<List<LogRecord>> future = SlaveLogFetcher.getLogRecords(computer.getChannel());
+                                                records = future.get(REMOTE_OPERATION_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+                                            }
+                                        }
+
+                                        if (records == null) {
```

Comment:
so recorderd could be null as the computer has no channel - so this will now be called - whereas before it was not.  
I can;t tell if this is intentional or not.